src/middlewared/middlewared/etc_files/smb_configure.py

```diff
@@ -376,43 +270,41 @@ def setup_samba_dirs(middleware, conf):
 
     if db_SID != domain_SID:
         logger.debug(f"Domain SID in group_mapping.tdb ({domain_SID}) is not SID in nas config ({db_SID}). Updating db")
-        ret = await set_database_SID(middleware, conf, domain_SID)
+        ret = set_database_SID(middleware, conf, domain_SID)
         if not ret:
             return ret
-        ret = await set_system_SID("setlocalsid", domain_SID)
+        ret = set_system_SID("setlocalsid", domain_SID)
 
     return ret
 
 
-async def validate_group_mappings(middleware, conf):
-    groupmap = await middleware.call('notifier.groupmap_list')
+def validate_group_mappings(middleware, conf):
+    groupmap = middleware.call_sync('smb.groupmap_list')
     if groupmap:
-        sids_fixed = await fixsid(middleware, conf, groupmap)
+        sids_fixed = fixsid(middleware, conf, groupmap)
         if not sids_fixed:
             groupmap = []
-    groups = await get_groups(middleware)
+    groups = get_groups(middleware)
     for g in groups:
-        if not await is_disallowed_group(middleware, conf, groupmap, g):
-            ret = await groupmap_add(conf, g, g)
-            if not ret:
-                logger.debug(f'Failed to generate group mapping for {g}')
+        if not any(filter(lambda x: g.upper() == x['ntgroup'].upper(), groupmap)):
+            middleware.call_sync('smb.groupmap_add', g)
 
 
-async def render(service, middleware):
+def render(service, middleware):
     conf = {}
-    conf = await get_config(middleware)
+    conf = get_config(middleware)
     if conf['systemdataset']['path'] is None:
         logger.debug("systemdataset.config returned 'None' as dataset path. Possible zpool import in progress. Exiting configure.")
         return
 
-    ret = await middleware.run_in_thread(setup_samba_dirs, middleware, conf)
+    ret = middleware.run_in_thread(setup_samba_dirs, middleware, conf)
```

Comment:
render is not a coroutine anymore so this wont work anymore, you should just run `setup_samba_dirs(middleware, conf)`